### ISSUE TITLE
Foot scratchpad issues

### ISSUE THREAD

**Issue by justinesmithies** [hyprwm/Hyprland#2078]
I'm using the latest git master on Arch

Recently if I add a foot terminal to the scratchpad then open the scratchpad revealing the foot terminal i can see parts of the wallpaper flashing through the terminal even though it's over the top of my qutebrowser until i move the mouse.
Then the next issue with the same is if i type exit in that terminal that is in the scratchpad i exit Hyprland back to my tty.
This doesn't happen if i type exit in a foot terminal that is not in the scratchpad.

Crash log:

```
--------------------------------------------
   Hyprland Crash Report
--------------------------------------------
Vaxry is going to be upset.

Hyprland received signal 11 (Segmentation fault)

Version: 6e58428336e55cf218812729ed99652d80f6f19b

System info:
	System name: Linux
	Node name: justine-laptop
	Release: 6.2.11-arch1-1
	Version: #1 SMP PREEMPT_DYNAMIC Thu, 13 Apr 2023 16:59:24 +0000

GPU:
	07:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Renoir [1002:1636] (rev d1) (prog-if 00 [VGA controller])


os-release:
	NAME="Arch Linux"
	PRETTY_NAME="Arch Linux"
	ID=arch
	BUILD_ID=rolling
	ANSI_COLOR="38;2;23;147;209"
	HOME_URL="https://archlinux.org/"
	DOCUMENTATION_URL="https://wiki.archlinux.org/"
	SUPPORT_URL="https://bbs.archlinux.org/"
	BUG_REPORT_URL="https://bugs.archlinux.org/"
	PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
	LOGO=archlinux-logo
	


Backtrace:
	#0 | Hyprland(_ZN13CrashReporter18createAndSaveCrashEi+0x831) [0x55f32d365941]
		??
		??:0
	#1 | Hyprland(_Z25handleUnrecoverableSignali+0x3c) [0x55f32d30169c]
		??
		??:0
	#2 | /usr/lib/libc.so.6(+0x38f50) [0x7fdb9ea51f50]
		??
		??:0
	#3 | Hyprland(_ZN13CHyprRenderer28renderAllClientsForWorkspaceEP8CMonitorP10CWorkspaceP8timespecRK8Vector2DRKf+0x843) [0x55f32d41b0c3]
		??
		??:0
	#4 | Hyprland(_ZN13CHyprRenderer15renderWorkspaceEP8CMonitorP10CWorkspaceP8timespecRK7wlr_box+0xf9) [0x55f32d41b4c9]
		??
		??:0
	#5 | Hyprland(_ZN13CHyprRenderer13renderMonitorEP8CMonitor+0xbbf) [0x55f32d41c59f]
		??
		??:0
	#6 | Hyprland(_Z13handleWrappedP11wl_listenerPv+0x3c) [0x55f32d3a1ddc]
		??
		??:0
	#7 | /usr/lib/libwayland-server.so.0(wl_signal_emit_mutable+0x7e) [0x7fdb9f4e401e]
		??
		??:0
	#8 | /usr/lib/libwlroots.so.12032(+0x51d08) [0x7fdb9f62dd08]
		??
		??:0
	#9 | /usr/lib/libdrm.so.2(drmHandleEvent+0xff) [0x7fdb9f4b59af]
		??
		??:0
	#10 | /usr/lib/libwlroots.so.12032(+0x54539) [0x7fdb9f630539]
		??
		??:0
	#11 | /usr/lib/libwayland-server.so.0(wl_event_loop_dispatch+0xa2) [0x7fdb9f4e5ae2]
		??
		??:0
	#12 | /usr/lib/libwayland-server.so.0(wl_display_run+0x27) [0x7fdb9f4e62d7]
		??
		??:0
	#13 | Hyprland(main+0x49b) [0x55f32d2f2bcb]
		??
		??:0
	#14 | /usr/lib/libc.so.6(+0x23790) [0x7fdb9ea3c790]
		??
		??:0
	#15 | /usr/lib/libc.so.6(__libc_start_main+0x8a) [0x7fdb9ea3c84a]
		??
		??:0
	#16 | Hyprland(_start+0x25) [0x55f32d301545]
		??
		??:0


Log tail:
[LOG] Searching for matching rules for org.qutebrowser.qutebrowser (title: [100%] JustineSmithies/hyprland-dotfiles: Dotfiles for my Hyprland setup with Eww bar - hyprland-dotfiles - Codeberg.org - qutebrowser)
[LOG] Window 302202e0 set title to [100%] JustineSmithies/hyprland-dotfiles: Dotfiles for my Hyprland setup with Eww bar - hyprland-dotfiles - Codeberg.org - qutebrowser
[LOG] Searching for matching rules for org.qutebrowser.qutebrowser (title: JustineSmithies/hyprland-dotfiles: Dotfiles for my Hyprland setup with Eww bar - hyprland-dotfiles - Codeberg.org - qutebrowser)
[LOG] Window 302202e0 set title to JustineSmithies/hyprland-dotfiles: Dotfiles for my Hyprland setup with Eww bar - hyprland-dotfiles - Codeberg.org - qutebrowser
[LOG] Keybind triggered, calling dispatcher (65, , 97)
[LOG] OPENINGON: 301e3ea0, Workspace: -99, Monitor: 0
[ERR] Orphaned Node 301d2dd0 (workspace ID: -99)!!
[LOG] Set keyboard focus to surface 30227ff0, with window name: justine@justine-laptop:~
[LOG] Showing the cursor (timeout)
[LOG] Set keyboard focus to surface 2ff7fad0, with window name: JustineSmithies/hyprland-dotfiles: Dotfiles for my Hyprland setup with Eww bar - hyprland-dotfiles - Codeberg.org - qutebrowser
[LOG] Keybind triggered, calling dispatcher (64, , 97)
[LOG] Toggling special workspace 804225952 to open
[LOG] Set keyboard focus to surface 301dd7d0, with window name: justine@justine-laptop:~
[LOG] Hiding the cursor (timeout)
[LOG] Showing the cursor (timeout)
[LOG] Window 301f0ad0 unmapped (class foot)
[LOG]  [hookSystem] New hook event registered: closeWindow
[LOG] Unregistered late callbacks XDG
[LOG] Callback 301f0af8 -> 301f0af0, XDG Window Late removed.
[LOG] Callback 301f0c98 -> 301f0c90, XDG Window Late removed.
[LOG] Callback 301f0dd0 -> 301f0dc8, XDG Window Late removed.
[LOG] Callback 301f0f08 -> 301f0f00, XDG Window Late removed.
[LOG] Callback 301f0ea0 -> 301f0e98, XDG Window Late removed.
[LOG] Callback 301f0e38 -> 301f0e30, XDG Window Late removed.
[LOG] Callback 301f0f70 -> 301f0f68, XDG Window Late removed.
[LOG] Callback 301f0d68 -> 301f0d60, XDG Window Late removed.
[LOG] Framebuffer created, status 36053
[LOG] Removing last node (dwindle)
[LOG] On closed window, new focused candidate is 302202e0
[LOG] Set keyboard focus to surface 2ff7fad0, with window name: JustineSmithies/hyprland-dotfiles: Dotfiles for my Hyprland setup with Eww bar - hyprland-dotfiles - Codeberg.org - qutebrowser
[LOG] Destroying the SubSurface tree of unmapped window 301f0ad0
[LOG] Callback 30256be0 -> 30256bd8, Subsurface removed.
[LOG] Callback 30256b10 -> 30256b08, Subsurface removed.
[LOG] Callback 30256b78 -> 30256b70, Subsurface removed.
[LOG] Callback 300cf038 -> 300cf030, SurfaceTreeNode removed.
[LOG] Callback 300cf0a0 -> 300cf098, SurfaceTreeNode removed.
[LOG] Callback 300cefd0 -> 300cefc8, SurfaceTreeNode removed.
[LOG] SurfaceTree Node removed
[LOG] Callback 301f1110 -> 301f1108, Toplevel removed.
[LOG] Callback 301f10a8 -> 301f10a0, Toplevel removed.
[LOG] Callback 301f1178 -> 301f1170, Toplevel removed.
[LOG] Callback 301f1250 -> 301f1248, CWLSurface removed.
[LOG] CWLSurface 301f1220 called destroy()
[LOG] Callback 301f0bc8 -> 301f0bc0, CWindow removed.
[LOG] Window 301f0ad0 destroyed, queueing. (class )
[LOG] Callback 301f0b60 -> 301f0b58, XDG Window removed.
[LOG] Callback 301f0c30 -> 301f0c28, XDG Window removed.
[ERR] onWindowRemovedTiling node null?
[LOG] Destroying workspace ID -99
[LOG]  [hookSystem] New hook event registered: destroyWorkspace
```

**Comment by vaxerski** [comment]
re: crash
check with c62ab1bee72cc4bda16a0574a8c9de7d71da314a

**Comment by justinesmithies** [comment]
Updated to the above and typing exit in the foot terminal that is in the scratchpad still exits Hyprland

```
--------------------------------------------
   Hyprland Crash Report
--------------------------------------------
Oops

Hyprland received signal 11 (Segmentation fault)

Version: c62ab1bee72cc4bda16a0574a8c9de7d71da314a

System info:
	System name: Linux
	Node name: justine-laptop
	Release: 6.2.11-arch1-1
	Version: #1 SMP PREEMPT_DYNAMIC Thu, 13 Apr 2023 16:59:24 +0000

GPU:
	07:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Renoir [1002:1636] (rev d1) (prog-if 00 [VGA controller])


os-release:
	NAME="Arch Linux"
	PRETTY_NAME="Arch Linux"
	ID=arch
	BUILD_ID=rolling
	ANSI_COLOR="38;2;23;147;209"
	HOME_URL="https://archlinux.org/"
	DOCUMENTATION_URL="https://wiki.archlinux.org/"
	SUPPORT_URL="https://bbs.archlinux.org/"
	BUG_REPORT_URL="https://bugs.archlinux.org/"
	PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
	LOGO=archlinux-logo
	


Backtrace:
	#0 | Hyprland(_ZN13CrashReporter18createAndSaveCrashEi+0x831) [0x556fafe2e871]
		??
		??:0
	#1 | Hyprland(_Z25handleUnrecoverableSignali+0x3c) [0x556fafdca63c]
		??
		??:0
	#2 | /usr/lib/libc.so.6(+0x38f50) [0x7fb921f69f50]
		??
		??:0
	#3 | Hyprland(_ZN13CHyprRenderer28renderAllClientsForWorkspaceEP8CMonitorP10CWorkspaceP8timespecRK8Vector2DRKf+0x843) [0x556fafee3e33]
		??
		??:0
	#4 | Hyprland(_ZN13CHyprRenderer15renderWorkspaceEP8CMonitorP10CWorkspaceP8timespecRK7wlr_box+0xf9) [0x556fafee4239]
		??
		??:0
	#5 | Hyprland(_ZN13CHyprRenderer13renderMonitorEP8CMonitor+0xbbf) [0x556fafee530f]
		??
		??:0
	#6 | Hyprland(_Z13handleWrappedP11wl_listenerPv+0x3c) [0x556fafe6ab9c]
		??
		??:0
	#7 | /usr/lib/libwayland-server.so.0(wl_signal_emit_mutable+0x7e) [0x7fb922a2001e]
		??
		??:0
	#8 | /usr/lib/libwlroots.so.12032(+0x51d08) [0x7fb922b69d08]
		??
		??:0
	#9 | /usr/lib/libdrm.so.2(drmHandleEvent+0xff) [0x7fb9229f19af]
		??
		??:0
	#10 | /usr/lib/libwlroots.so.12032(+0x54539) [0x7fb922b6c539]
		??
		??:0
	#11 | /usr/lib/libwayland-server.so.0(wl_event_loop_dispatch+0xa2) [0x7fb922a21ae2]
		??
		??:0
	#12 | /usr/lib/libwayland-server.so.0(wl_display_run+0x27) [0x7fb922a222d7]
		??
		??:0
	#13 | Hyprland(main+0x49b) [0x556fafdbbb6b]
		??
		??:0
	#14 | /usr/lib/libc.so.6(+0x23790) [0x7fb921f54790]
		??
		??:0
	#15 | /usr/lib/libc.so.6(__libc_start_main+0x8a) [0x7fb921f5484a]
		??
		??:0
	#16 | Hyprland(_start+0x25) [0x556fafdca4e5]
		??
		??:0


Log tail:
[LOG] Window b2eb76f0 set title to justine@justine-laptop:~
[LOG] Searching for matching rules for org.qutebrowser.qutebrowser (title: [100%] hyprwm/Hyprland: Hyprland is a highly customizable dynamic tiling Wayland compositor that doesn't sacrifice on its looks. - qutebrowser)
[LOG] Window b2e924d0 set title to [100%] hyprwm/Hyprland: Hyprland is a highly customizable dynamic tiling Wayland compositor that doesn't sacrifice on its looks. - qutebrowser
[LOG] Searching for matching rules for org.qutebrowser.qutebrowser (title: hyprwm/Hyprland: Hyprland is a highly customizable dynamic tiling Wayland compositor that doesn't sacrifice on its looks. - qutebrowser)
[LOG] Window b2e924d0 set title to hyprwm/Hyprland: Hyprland is a highly customizable dynamic tiling Wayland compositor that doesn't sacrifice on its looks. - qutebrowser
[LOG] Keybind triggered, calling dispatcher (65, , 97)
[LOG] OPENINGON: b2ea9610, Workspace: -99, Monitor: 0
[ERR] Orphaned Node b2ee6430 (workspace ID: -99)!!
[LOG] Set keyboard focus to surface b2f802f0, with window name: justine@justine-laptop:~
[LOG] Keybind triggered, calling dispatcher (68, , 46)
[LOG] Set keyboard focus to surface b2d5f1b0, with window name: hyprwm/Hyprland: Hyprland is a highly customizable dynamic tiling Wayland compositor that doesn't sacrifice on its looks. - qutebrowser
[LOG] Keybind triggered, calling dispatcher (64, , 97)
[LOG] Toggling special workspace -1295180288 to open
[LOG] Set keyboard focus to surface b2eb6f00, with window name: justine@justine-laptop:~
[LOG] Window b2eb76f0 unmapped (class foot)
[LOG]  [hookSystem] New hook event registered: closeWindow
[LOG] Unregistered late callbacks XDG
[LOG] Callback b2eb7718 -> b2eb7710, XDG Window Late removed.
[LOG] Callback b2eb78b8 -> b2eb78b0, XDG Window Late removed.
[LOG] Callback b2eb79f0 -> b2eb79e8, XDG Window Late removed.
[LOG] Callback b2eb7b28 -> b2eb7b20, XDG Window Late removed.
[LOG] Callback b2eb7ac0 -> b2eb7ab8, XDG Window Late removed.
[LOG] Callback b2eb7a58 -> b2eb7a50, XDG Window Late removed.
[LOG] Callback b2eb7b90 -> b2eb7b88, XDG Window Late removed.
[LOG] Callback b2eb7988 -> b2eb7980, XDG Window Late removed.
[LOG] Framebuffer created, status 36053
[LOG] Removing last node (dwindle)
[LOG] On closed window, new focused candidate is b2e924d0
[LOG] Set keyboard focus to surface b2d5f1b0, with window name: hyprwm/Hyprland: Hyprland is a highly customizable dynamic tiling Wayland compositor that doesn't sacrifice on its looks. - qutebrowser
[LOG] Destroying the SubSurface tree of unmapped window b2eb76f0
[LOG] Callback b2eba310 -> b2eba308, Subsurface removed.
[LOG] Callback b2eba240 -> b2eba238, Subsurface removed.
[LOG] Callback b2eba2a8 -> b2eba2a0, Subsurface removed.
[LOG] Callback b2ea3cb8 -> b2ea3cb0, SurfaceTreeNode removed.
[LOG] Callback b2ea3d20 -> b2ea3d18, SurfaceTreeNode removed.
[LOG] Callback b2ea3c50 -> b2ea3c48, SurfaceTreeNode removed.
[LOG] SurfaceTree Node removed
[LOG] Callback b2eb7d30 -> b2eb7d28, Toplevel removed.
[LOG] Callback b2eb7cc8 -> b2eb7cc0, Toplevel removed.
[LOG] Callback b2eb7d98 -> b2eb7d90, Toplevel removed.
[LOG] Callback b2eb7e70 -> b2eb7e68, CWLSurface removed.
[LOG] CWLSurface b2eb7e40 called destroy()
[LOG] Callback b2eb77e8 -> b2eb77e0, CWindow removed.
[LOG] Set keyboard focus to surface b2f802f0, with window name: justine@justine-laptop:~
[LOG] Destroying workspace ID -99
[LOG]  [hookSystem] New hook event registered: destroyWorkspace
[LOG] Window b2eb76f0 destroyed, queueing. (class )
[LOG] Callback b2eb7780 -> b2eb7778, XDG Window removed.
[LOG] Callback b2eb7850 -> b2eb7848, XDG Window removed.
[ERR] onWindowRemovedTiling node null?
```

**Comment by justinesmithies** [comment]
Right I can reproduce it now but I am using an external screen and my laptops. It seems if I open two terminals on the laptop then put one into the scratchpad and open the scratchpad on the other screen when I type exit into the terminal it quits Hyprland. Where as if it was just a terminal moved to that screen typing exit just quits the terminal as expected.

My dots are [here](https://codeberg.org/JustineSmithies/hyprland-dotfiles) as I have some scripts setup to make workspaces work how I feel comfortable coming from Qtile.


Logs

```
--------------------------------------------
   Hyprland Crash Report
--------------------------------------------
I tried so hard, and got so far...

Hyprland received signal 11 (Segmentation fault)

Version: c62ab1bee72cc4bda16a0574a8c9de7d71da314a

System info:
	System name: Linux
	Node name: justine-laptop
	Release: 6.2.11-arch1-1
	Version: #1 SMP PREEMPT_DYNAMIC Thu, 13 Apr 2023 16:59:24 +0000

GPU:
	07:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Renoir [1002:1636] (rev d1) (prog-if 00 [VGA controller])


os-release:
	NAME="Arch Linux"
	PRETTY_NAME="Arch Linux"
	ID=arch
	BUILD_ID=rolling
	ANSI_COLOR="38;2;23;147;209"
	HOME_URL="https://archlinux.org/"
	DOCUMENTATION_URL="https://wiki.archlinux.org/"
	SUPPORT_URL="https://bbs.archlinux.org/"
	BUG_REPORT_URL="https://bugs.archlinux.org/"
	PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
	LOGO=archlinux-logo
	


Backtrace:
	#0 | Hyprland(_ZN13CrashReporter18createAndSaveCrashEi+0x831) [0x55e37f7a0871]
		??
		??:0
	#1 | Hyprland(_Z25handleUnrecoverableSignali+0x3c) [0x55e37f73c63c]
		??
		??:0
	#2 | /usr/lib/libc.so.6(+0x38f50) [0x7f8dd2651f50]
		??
		??:0
	#3 | Hyprland(_ZN13CHyprRenderer28renderAllClientsForWorkspaceEP8CMonitorP10CWorkspaceP8timespecRK8Vector2DRKf+0x843) [0x55e37f855e33]
		??
		??:0
	#4 | Hyprland(_ZN13CHyprRenderer15renderWorkspaceEP8CMonitorP10CWorkspaceP8timespecRK7wlr_box+0xf9) [0x55e37f856239]
		??
		??:0
	#5 | Hyprland(_ZN13CHyprRenderer13renderMonitorEP8CMonitor+0xbbf) [0x55e37f85730f]
		??
		??:0
	#6 | Hyprland(_Z13handleWrappedP11wl_listenerPv+0x3c) [0x55e37f7dcb9c]
		??
		??:0
	#7 | /usr/lib/libwayland-server.so.0(wl_signal_emit_mutable+0x7e) [0x7f8dd316e01e]
		??
		??:0
	#8 | /usr/lib/libwayland-server.so.0(wl_event_loop_dispatch_idle+0x23) [0x7f8dd316fa23]
		??
		??:0
	#9 | /usr/lib/libwayland-server.so.0(wl_event_loop_dispatch+0xfd) [0x7f8dd316fb3d]
		??
		??:0
	#10 | /usr/lib/libwayland-server.so.0(wl_display_run+0x27) [0x7f8dd31702d7]
		??
		??:0
	#11 | Hyprland(main+0x49b) [0x55e37f72db6b]
		??
		??:0
	#12 | /usr/lib/libc.so.6(+0x23790) [0x7f8dd263c790]
		??
		??:0
	#13 | /usr/lib/libc.so.6(__libc_start_main+0x8a) [0x7f8dd263c84a]
		??
		??:0
	#14 | Hyprland(_start+0x25) [0x55e37f73c4e5]
		??
		??:0


Log tail:
[ERR] onWindowRemovedTiling node null?
[LOG] Cleanup: destroyed a window
[LOG] Keybind triggered, calling dispatcher (65, , 97)
[LOG] Removing last node (dwindle)
[LOG] OPENINGON: 0, Workspace: -99, Monitor: 1
[ERR] Orphaned Node 80e2e190 (workspace ID: -99)!!
[LOG] Keybind triggered, calling dispatcher (65, , 50)
[LOG] Keybind triggered, calling dispatcher (64, , 97)
[LOG] Toggling special workspace -2135067904 to open
[LOG] Set keyboard focus to surface 80e37030, with window name: justine@justine-laptop:~
[LOG] Keybind triggered, calling dispatcher (65, , 50)
[LOG] Removing last node (dwindle)
[LOG] OPENINGON: 80e2e190, Workspace: 2, Monitor: 0
[LOG] Destroying workspace ID -99
[LOG] Keybind triggered, calling dispatcher (65, , 97)
[LOG] Removing last node (dwindle)
[LOG] OPENINGON: 0, Workspace: -99, Monitor: 0
[ERR] Orphaned Node 8165a900 (workspace ID: -99)!!
[LOG] Keybind triggered, calling dispatcher (68, , 46)
[LOG] Keybind triggered, calling dispatcher (64, , 97)
[LOG] Toggling special workspace -2135067904 to open
[LOG] Set keyboard focus to surface 80e37030, with window name: justine@justine-laptop:~
[LOG] Window 81662380 unmapped (class foot)
[LOG] Unregistered late callbacks XDG
[LOG] Callback 816623a8 -> 816623a0, XDG Window Late removed.
[LOG] Callback 81662548 -> 81662540, XDG Window Late removed.
[LOG] Callback 81662680 -> 81662678, XDG Window Late removed.
[LOG] Callback 816627b8 -> 816627b0, XDG Window Late removed.
[LOG] Callback 81662750 -> 81662748, XDG Window Late removed.
[LOG] Callback 816626e8 -> 816626e0, XDG Window Late removed.
[LOG] Callback 81662820 -> 81662818, XDG Window Late removed.
[LOG] Callback 81662618 -> 81662610, XDG Window Late removed.
[LOG] Framebuffer created, status 36053
[LOG] Removing last node (dwindle)
[LOG] On closed window, new focused candidate is 0
[LOG] Destroying the SubSurface tree of unmapped window 81662380
[LOG] Callback 80e244b0 -> 80e244a8, Subsurface removed.
[LOG] Callback 80e243e0 -> 80e243d8, Subsurface removed.
[LOG] Callback 80e24448 -> 80e24440, Subsurface removed.
[LOG] Callback 80e2a518 -> 80e2a510, SurfaceTreeNode removed.
[LOG] Callback 80e2a580 -> 80e2a578, SurfaceTreeNode removed.
[LOG] Callback 80e2a4b0 -> 80e2a4a8, SurfaceTreeNode removed.
[LOG] SurfaceTree Node removed
[LOG] Callback 816629c0 -> 816629b8, Toplevel removed.
[LOG] Callback 81662958 -> 81662950, Toplevel removed.
[LOG] Callback 81662a28 -> 81662a20, Toplevel removed.
[LOG] Callback 81662b00 -> 81662af8, CWLSurface removed.
[LOG] CWLSurface 81662ad0 called destroy()
[LOG] Callback 81662478 -> 81662470, CWindow removed.
[LOG] Destroying workspace ID -99
```
